### PR TITLE
add box component

### PR DESCRIPTION
**Pull request by kevanantha**
here's the examples
image

P.S the code a bit messy (fix it later), I just need feedback about the structure app

### CODE REVIEW

**Comment by faultable** [comment]
bapak @ri7nz sang ahli design system sebagai reviewer project ini 🙏

**Comment by kevanantha** [comment]
bapak @ri7nz

File: examples/cra/src/index.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+body {
```
Comment (ri7nz) [inline examples/cra/src/index.css]: Please Remove `index.css`, we need to use font from styled-system

File: examples/cra/src/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
```
Comment (ri7nz) [inline examples/cra/src/index.js]: remove it

File: packages/edan-core/package.json [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+  "main": "src/index.js",
   "repository": "https://github.com/evilfactorylabs/edan",
   "author": "evilfactory",
   "license": "MIT",
   "dependencies": {
-    "@emotion/core": "^10.0.27",
+    "@emotion/core": "^10.0.28",
     "@emotion/styled": "^10.0.27",
+    "@styled-system/should-forward-prop": "^5.1.5",
     "emotion-theming": "^10.0.27",
```
Comment (ri7nz) [inline packages/edan-core/package.json]: we need `@styled-system/props-types`

File: packages/edan-core/src/box/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  layout
+)
+
+const Box = styled('div', { shouldForwardProp })(composeStyles, {
+  boxSizing: 'border-box',
+  minWidth: 0
+})
+
+Box.displayName = 'Box'
+
```
Comment (ri7nz) [inline packages/edan-core/src/box/index.js]: Good job, LGTM @kevanantha

File: examples/cra/src/App.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import logo from './logo.svg'
+import './App.css'
+import { Box } from '@evilfactory/edan-core'
+
+function App() {
+  return (
+    <>
```
Comment (ri7nz) [inline examples/cra/src/App.js]: Add emotion themeprovider and value of provider use from src/theme/index.js#default

Comment (ri7nz) [reply examples/cra/src/App.js]: in `edan core` add `src/ThemeProvider/index.js`
```javascript
import { ThemeProvider as Provider } from 'emotion-theming';
import { default as defaultTheme } from '../theme';


export default function ThemeProvider({children, theme}){
  return <Provider theme={theme}>{children}</Provider>;
}

ThemeProvider.defaultProps = {
  theme: defaultTheme
};
// proptypes
```

in `cra`, implement ThemeProvider in `App.js`.

@kevanantha

File: examples/cra/src/index.js (hunk 2 above)
Comment (faultable) [inline examples/cra/src/index.js]:
```suggestion
```

begindang

Comment (kevanantha) [reply examples/cra/src/index.js]: nanti runnya gimana klo reactDOMnya di hapus?

File: packages/edan-core/src/box/index.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  border,
+  color,
+  compose,
+  flexbox,
+  grid,
+  layout,
+  position,
+  shadow,
+  space,
+  typography
```
Comment (ri7nz) [inline packages/edan-core/src/box/index.js]:
```suggestion
  typography,
  rdii
```

File: packages/edan-core/src/box/index.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  space,
+  color,
+  typography,
+  flexbox,
+  grid,
+  background,
+  position,
+  shadow,
+  border,
+  layout
```
Comment (ri7nz) [inline packages/edan-core/src/box/index.js]:
```suggestion
  layout,
  rdii
```
for borderRadius

File: examples/cra/src/index.js (hunk 2 above)
Comment (kevanantha) [reply examples/cra/src/index.js]:
```suggestion
```

File: examples/cra/src/index.css [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+    sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+    monospace;
+}
```
Comment (ri7nz) [inline examples/cra/src/index.css]:
```suggestion

```
Remove :rocket:

File: examples/cra/src/index.js (hunk 2 above)
Comment (ri7nz) [reply examples/cra/src/index.js]: @kevanantha  slow, ini contoh aja dari @faultable

File: packages/edan-core/src/box/index.js (hunk 7 above)
Comment (kevanantha) [reply packages/edan-core/src/box/index.js]: do you wanna use `rdii` for naming conversion? @ri7nz

**Comment by ri7nz** [comment]
image

**Pull request merged**